src/libraries/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -518,12 +518,15 @@ public static void DeserializeUnsupportedType()
             // Any test payload is fine.
             string json = @"""Some string""";
 
-            RunTest<Type>();
-            RunTest<SerializationInfo>();
+            RunTest<Type>(json);
+            RunTest<SerializationInfo>(json);
+            RunTest<IntPtr>(json);
+            RunTest<UIntPtr>(json);
 
-            void RunTest<T>()
+            void RunTest<T>(string json)
             {
-                string fullName = typeof(T).FullName;
+                Type type = typeof(T);
```

Comment:
Nit: I would personally skip this change and inline a `typeof(T)` call in line 544.